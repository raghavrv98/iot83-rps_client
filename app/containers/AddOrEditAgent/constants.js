/*
 *
 * AddOrEditAgent constants
 *
 */

export const CREATE_AGENT_REQUEST = "app/AddOrEditAgent/CREATE_AGENT_REQUEST";
export const CREATE_AGENT_SUCCESS = "app/AddOrEditAgent/CREATE_AGENT_SUCCESS";
export const CREATE_AGENT_FAILURE = "app/AddOrEditAgent/CREATE_AGENT_FAILURE";

export const GET_AGENT_DETAILS = "app/AddOrEditAgent/GET_AGENT_DETAILS";
export const GET_AGENT_DETAILS_SUCCESS = "app/AddOrEditAgent/GET_AGENT_DETAILS_SUCCESS";
export const GET_AGENT_DETAILS_FAILURE = "app/AddOrEditAgent/GET_AGENT_DETAILS_FAILURE";
