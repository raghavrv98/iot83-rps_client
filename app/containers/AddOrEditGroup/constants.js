/*
 *
 * AddOrEditGroup constants
 *
 */

export const ON_SUBMIT_REQUEST = "app/AddOrEditGroup/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_FAILURE";

export const GET_GROUP_DETAILS = "app/AddOrEditGroup/GET_GROUP_DETAILS";
export const GET_GROUP_DETAILS_SUCCESS = "app/AddOrEditGroup/GET_GROUP_DETAILS_SUCCESS";
export const GET_GROUP_DETAILS_FAILURE = "app/AddOrEditGroup/GET_GROUP_DETAILS_FAILURE";
