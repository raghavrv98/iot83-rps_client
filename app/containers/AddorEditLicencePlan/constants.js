/*
 *
 * AddorEditLicencePlan constants
 *
 */

export const DEFAULT_ACTION = "app/AddorEditLicencePlan/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_FAILURE";

export const GET_PLANS_LIST = "app/AddorEditLicencePlan/GET_PLANS_LIST";
export const GET_PLANS_LIST_SUCCESS = "app/AddorEditLicencePlan/GET_PLANS_LIST_SUCCESS";
export const GET_PLANS_LIST_FAILURE = "app/AddorEditLicencePlan/GET_PLANS_LIST_FAILURE";