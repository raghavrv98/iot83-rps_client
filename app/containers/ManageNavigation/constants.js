/*
 *
 * ManageNavigation constants
 *
 */

export const DEFAULT_ACTION = "app/ManageNavigation/DEFAULT_ACTION";

export const GET_MENU_REQUEST = "app/ManageNavigation/GET_MENU_REQUEST";
export const GET_MENU_SUCCESS = "app/ManageNavigation/GET_MENU_SUCCESS";
export const GET_MENU_FAILURE = "app/ManageNavigation/GET_MENU_FAILURE";

export const MENU_SAVE_HANDLER = "app/ManageNavigation/MENU_SAVE_HANDLER";
export const MENU_SAVE_HANDLER_SUCCESS = "app/ManageNavigation/MENU_SAVE_HANDLER_SUCCESS";
export const MENU_SAVE_HANDLER_FAILURE = "app/ManageNavigation/MENU_SAVE_HANDLER_FAILURE";
