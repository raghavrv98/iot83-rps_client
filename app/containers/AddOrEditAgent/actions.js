/*
 *
 * AddOrEditAgent actions
 *
 */

import { CREATE_AGENT_REQUEST,GET_AGENT_DETAILS } from "./constants";

export function createAgent(data) {
  return {
    type: CREATE_AGENT_REQUEST,
    payload: data,
  };
}
export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}
