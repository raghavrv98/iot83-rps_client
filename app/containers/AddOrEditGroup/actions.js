/*
 *
 * AddOrEditGroup actions
 *
 */

import { ON_SUBMIT_REQUEST,GET_GROUP_DETAILS } from "./constants";

export function onSubmitHandler(payload,id) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    id
  };
}

export function getGroupDetails(id) {
  return {
    type: GET_GROUP_DETAILS,
    id
  };
}
