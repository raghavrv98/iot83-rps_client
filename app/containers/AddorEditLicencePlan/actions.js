/*
 *
 * AddorEditLicencePlan actions
 *
 */

import { DEFAULT_ACTION, ON_SUBMIT_REQUEST, GET_PLANS_LIST } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function onSubmitHandler(payload) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload
  };
}

export function getPlansList(){
  return{
    type: GET_PLANS_LIST
  }
}