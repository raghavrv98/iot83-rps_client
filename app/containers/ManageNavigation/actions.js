/*
 *
 * ManageNavigation actions
 *
 */

import { DEFAULT_ACTION, GET_MENU_REQUEST,MENU_SAVE_HANDLER } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function getMenu() {
  return {
    type: GET_MENU_REQUEST
  };
}

export function navlistSaveHandler(payload) {
  return {
    type: MENU_SAVE_HANDLER,
    payload
  };
}
