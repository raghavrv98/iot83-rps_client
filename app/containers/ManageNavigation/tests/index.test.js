import React, { Component } from 'react';
import { mount, shallow } from 'enzyme';
import history from "utils/history";
import { getMenu, navlistSaveHandler } from "../actions";
import { GET_MENU_REQUEST, MENU_SAVE_HANDLER } from "../constants";
import { ManageNavigation, mapDispatchToProps } from "../index";

const navList = [
    {
      _id: "9bed0d5ca79011e9863a0242ac110016",
      plantId: "9bed0d5ca79011e9863a0242ac110016",
      type:"demo",
      childVisibility:false,
      posit: 1,
      subMenus:["demo"]
    }];

describe('< ManageNavigation />', () => {
    const match = {
        params: {
            payload: "demo"
        }
    };
    it('Expect to have unit tests specified', () => {
        expect(true).toEqual(true);
    });

    it('match snapshots', () => {
        const wrap = shallow(<ManageNavigation match={match} getMenu={getMenu} />)
        expect(wrap).toMatchSnapshot()
    })

    it("test componentWillMount", async () => {
        const fetchZonesList = jest.fn();
        spyOn(ManageNavigation.prototype, "componentDidMount").and.callThrough();
        shallow(
            <ManageNavigation
                match={match}
                getMenu={getMenu}
            />
        );
        expect(ManageNavigation.prototype.componentDidMount).toHaveBeenCalledTimes(1);
        expect(fetchZonesList).toHaveBeenCalledTimes(0);
    });

    it("Function testing on mapDispatchToProps", () => {
        const mockParams = {
          params: {
            payload: "demo"
          }
        };
        const dispatch = jest.fn();
        mapDispatchToProps(dispatch).getMenu();
        expect(dispatch.mock.calls[0][0]).toEqual({
          type: GET_MENU_REQUEST
        });
        mapDispatchToProps(dispatch).navlistSaveHandler(
            mockParams.params.payload
        );
        expect(dispatch.mock.calls[1][0]).toEqual({
          type: MENU_SAVE_HANDLER,
          payload: "demo"
        });
    });
});