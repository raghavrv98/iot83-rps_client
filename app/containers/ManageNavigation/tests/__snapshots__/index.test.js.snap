// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< ManageNavigation /> match snapshots 1`] = `
<div
  className="appContent"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      ManageNavigation
    </title>
    <meta
      content="Description of ManageNavigation"
      name="description"
    />
  </HelmetWrapper>
  <div
    className="pageBreadcrumb"
  >
    <div
      className="flex-item fx-b70"
    >
      <p>
        <FormattedMessage
          defaultMessage="Manage Navigations"
          id="app.containers.ManageNavigation.titleManageNav"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </p>
      <h5>
        <FormattedMessage
          defaultMessage="All Available Navigations"
          id="app.containers.ManageNavigation.title"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </h5>
    </div>
    <div
      className="flex-item fx-b30 text-right align-items-center"
    />
  </div>
</div>
`;
